core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -201,18 +201,35 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * If the add time has expired, request new executors and refresh the add time.
-   * If the remove time for an existing executor has expired, kill the executor.
+   * The number of executors we would have if the cluster manager were to fulfill all our requests.
+   */
+  private def targetNumExecutors(): Int =
+    numExecutorsPending + executorIds.size
```

Comment:
an earlier version of this subtracted the number of executors pending to be removed. Is there a reason why you took that out?